app/controllers/users_controller.rb

```diff
@@ -0,0 +1,25 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find_by id: params[:id]
```

Comment:
thêm code xử lý khi user bị nil như bên dưới nhé

`return if @user`
`flash[:warning] = "User not found"`
`redirect_to root_path`